internal/init/prompts.go

```diff
@@ -170,6 +200,17 @@ func sanitizeParameterValue(str string) string {
 
 // PromptParams renders series of prompt UI based on the config
 func PromptModuleParams(moduleConfig moduleconfig.ModuleConfig, parameters map[string]string, projectCredentials globalconfig.ProjectCredential) (map[string]string, error) {
+	// map module field names to corresponding validate functions.
+	// not optimal solution as changing a field name in the EKS stack would render the map invalid.
+	m := map[string]func(string) error{
+		"productionHostRoot": validateRootDomain,
+		"stagingHostRoot":    validateRootDomain,
+
+		"productionFrontendSubdomain": validateSubDomain,
+		"productionBackendSubdomain":  validateSubDomain,
+		"stagingFrontendSubdomain":    validateSubDomain,
+		"stagingBackendSubdomain":     validateSubDomain,
+	}
```

Comment:
I'm going to move validation to the module side, allowing module variables to specify a regex to pick against or a predefined list of validation functions scales better. good suggestion @davidcheung 💡